src/main/groovy/org/shipkit/internal/gradle/plugin/PluginValidatorTask.java

```diff
@@ -0,0 +1,39 @@
+package org.shipkit.internal.gradle.plugin;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.TaskAction;
+
+import java.io.File;
+import java.util.Set;
+
+
+public class PluginValidatorTask extends DefaultTask {
+
+    @InputFiles
+    private Set<File> gradlePlugins;
+    @InputFiles
+    private Set<File> gradleProperties;
+
+
+    @TaskAction
+    public void validate() {
+        new PluginValidator().validate(gradlePlugins, gradleProperties);
+    }
+
+    public void setGradlePlugins(Set<File> gradlePlugins) {
+        this.gradlePlugins = gradlePlugins;
```

Comment:
We may need to remove this public API from the task. See my previous comment about doing discovery in the task execution phase. You may want to keep some public API on the task, for example to specify the source set where we look for plugins.

Later on, we may need to add some public API (methods/setters) on the task so that we can exclude some plugins from validation). We will see if we need it ;)